### PR TITLE
Refactor instruct names behavior controls

### LINKED ISSUE #2819: [FEATURE_REQUEST] Adjust CC's Default names behavior and make TC's options consistent with CC
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

Currently, Chat Completion's "Default" is more sane and adds prefixes for groups and past personas. The problem is Text Completion's "Force for Groups and Personas" forces current persona even in solo chat ("Include Names" unchecked). In fact, this feels like a bug.

### Describe the solution you'd like

~~Change TC's "Force for Groups and Personas" to "Force for Groups and Past Personas". That's it, this will make TC's two checkboxes match CC's None (TC both unchecked), Default (TC 2nd checked), and Message Content (TC 1st checked, or 2nd checked in group chat).~~

Edit: Request changed slightly. Make the "past persona" part of CC's Default solo chat apply to group chat too.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Priority

Low (Nice-to-have)

### Are you willing to test this on staging/unstable branch if this is implemented?

Yes

### PR DESCRIPTION
**Pull request by Cohee1207**
Closes #2819

Splits "Include names" and "Force for groups" checkboxes into three radio buttons.

- No names added
- Names added in groups and for past personas
- Names added always

All existing settings and templates should migrate automatically.

*I did not yet test the functional part, only the migration.*

## Checklist:

- [X] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cloak1505** [comment]
```
Test cases (OK = expected):

            CC             TC
None        OK             OK
Def solo*   OK             P always on
Def group   P always on    P always on
Always      OK             OK

*/sendas name=OtherChar gets name in solo, OK
 CC solo shows past Persona, OK
 ```
 
 The UI looks good, but functionally persona names haven't changed.

**Comment by Cohee1207** [comment]
> The UI looks good, but functionally persona names haven't changed.

Yep, I did not change the functionality, only the UI.

Needs a fix cause the label doesn't match the behavior.

**Comment by Cohee1207** [comment]
@cloak1505 please check again. I updated the logic for chat history formatting. 
1. Message examples formatting is not changed. FORCE always adds the name of the current persona.
2. Last prompt line is not changed. Not a concern for regular generations, and impersonation only adds a name of the current persona on ALWAYS.

**Comment by cloak1505** [comment]
Getting closer.
```
           CC             TC
None       OK             OK
Def solo   OK             OK
Def group  P always on    P always on
Always     OK             OK
```

**Comment by Cohee1207** [comment]
Wait, so what's the expected result for groups?

**Comment by cloak1505** [comment]
> Wait, so what's the expected result for groups?

I'd like for it to apply past personas only, same way solo handles persona. Users who need current persona name can use Always.

I would find it strange if someone doesn't need names in solo but suddenly needs current persona in group chat. If not having current persona breaks a user's group chat (e.g. model not knowing who's speaking), shouldn't it break solo too?

Anyone like me who need current persona name off, currently don't have the option for group chars only.

I suppose "Groups and Past Personas" is ambiguous (under the requested change). "Group Chars and Past Personas" is clearer.

Back when I said that anything more complex would require more checkboxes and clutter the UI, I was referring to splitting solo and group chat behaviors. "Group Chars and Past Personas" strikes a balance between Never and Always and "inserts names when needed".

**Comment by Cohee1207** [comment]
You mean add group names only for chars and past personas, but not for users in forced (default mode)?

**Comment by cloak1505** [comment]
Current = [58b8c7c](https://github.com/SillyTavern/SillyTavern/pull/2837/commits/58b8c7c634d4db70d4e52bb30f2d3b3c9827a1ec)
Never = all names off, Always = all names on

| "Group Chars and Past Personas" option| Current Persona | Past Persona | {{char}} |
| - | - | - | - |
| solo chat, current and expected | off | on | off (`/sendas name=Other` on) |
| group chat, expected | ***off*** | on | on |
| group chat, current | ***on*** | on | on |

My problem is that currently the middle option in group chat is exactly the same as Always.

**Comment by Cohee1207** [comment]
@cloak1505 updated both CC and TC.

**Comment by cloak1505** [comment]
Nice. 👍

**Comment by Cohee1207** [comment]
@Wolfsblvt feel free to give it a final review since we're done with testing.

File: public/scripts/instruct-mode.js [hunk 1]
```diff
@@ -311,9 +329,9 @@ export const force_output_sequence = {
  * @returns {string} Formatted instruct mode chat message.
  */
 export function formatInstructModeChat(name, mes, isUser, isNarrator, forceAvatar, name1, name2, forceOutputSequence) {
-    let includeNames = isNarrator ? false : power_user.instruct.names;
+    let includeNames = isNarrator ? false : power_user.instruct.names_behavior === names_behavior_types.ALWAYS;
 
-    if (!isNarrator && power_user.instruct.names_force_groups && (selected_group || forceAvatar)) {
+    if (!isNarrator && power_user.instruct.names_behavior === names_behavior_types.FORCE && ((selected_group && name !== name1) || (forceAvatar && name !== name1))) {
```
Comment (Wolfsblvt) [inline public/scripts/instruct-mode.js:334]: Nice that you added this check for `name !== name1` on the `forceAvatar` **after** I fixed `/sendas` to not include forced avatars if the char is the same as the current chat :D

Though it makes sense to have both, especially for historic purposes. There is no need to enforce name based on forced avatar alone, if the name is the same as the char.

**Review by Wolfsblvt — APPROVED** [review]
I did not test functionality, as y'all already did the testing.

Code looks good, I didn't see anything weird or any issues.
I also made sure to check the default instruct templates for typos. All good too.

So, good to go.

(Why you decided to add ST's first selector (I think?) instead of a dropdown I don't know though. I don't mind - but it's a very weird looking special control now)

**Comment by Cohee1207** [comment]
> (Why you decided to add ST's first selector (I think?) instead of a dropdown I don't know though. I don't mind - but it's a very weird looking special control now)

I made it radio buttons for similarity with CC's similar control.

**Pull request merged**